MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -300,7 +294,46 @@ namespace WPEFramework {
                 IARM_CHECK(IARM_Bus_RegisterEventHandler(IARM_BUS_MAINTENANCE_MGR_NAME, IARM_BUS_MAINTENANCEMGR_EVENT_UPDATE, _MaintenanceMgrEventHandler));
                 //Register for setMaintenanceStartTime
                 IARM_CHECK(IARM_Bus_RegisterEventHandler(IARM_BUS_MAINTENANCE_MGR_NAME, IARM_BUS_DCM_NEW_START_TIME_EVENT,_MaintenanceMgrEventHandler));
-           }
+
+                maintenanceManagerOnBootup();
+            }
+        }
+
+        void MaintenanceManager::maintenanceManagerOnBootup() {
+            /* on boot up we set these things */
+            MaintenanceManager::g_currentMode = FOREGROUND_MODE;
+
+            MaintenanceManager::g_notify_status=MAINTENANCE_IDLE;
+            MaintenanceManager::g_epoch_time="";
+
+            MaintenanceManager::g_is_critical_maintenance="true";
+            MaintenanceManager::g_is_reboot_pending="";
+            MaintenanceManager::g_lastSuccessful_maint_time="";
+            MaintenanceManager::g_task_status=0;
+
+            /* we post just to tell that we are in idle at this moment */
+            MaintenanceManager::_instance->onMaintenanceStatusChange(g_notify_status);
+
+            int32_t exec_status=E_NOK;
+
+            /* we call dcmscript to get the new start time */
+            if (Utils::fileExists("/lib/rdk/StartDCM_maintaince.sh")) {
+                exec_status = system("/lib/rdk/StartDCM_maintaince.sh &");
+                if ( E_OK == exec_status ){
+                    LOGINFO("DBG:Succesfully executed StartDCM_maintaince.sh \n");
+                }
+                else {
+                    LOGINFO("DBG:Failed to execute StartDCM_maintaince.sh !! \n");
+                }
+            }
+            else {
+                LOGINFO("DBG: Unable to find StartDCM_maintaince.sh \n");
+            }
+
+            /* we moved every thing to a thread */
+            /* only when dcm is getting a DCM_SUCCESS/DCM_ERROR we say
+             * Maintenance is started until then we say MAITENANCE_IDLE */
+            m_thread = std::thread(&MaintenanceManager::task_execution_thread, _instance);
```

Comment:
m_thread needs to be joined during DeInitialize() or DeinitializeIARM(). You can if you want use Utils::ThreadRAII type for m_thread which will auto join on destruction.

Reply:
Added in DeInitialize, line 505. we will create a cleanup ticket to address other things.